misc/quick-protobuf-codec/src/lib.rs

```diff
@@ -31,21 +30,22 @@ impl<In, Out> Codec<In, Out> {
     }
 }
 
-impl<In: Message, Out> Encoder for Codec<In, Out> {
+impl<In: MessageWrite, Out> Encoder for Codec<In, Out> {
     type Item = In;
     type Error = Error;
 
     fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error> {
         let mut encoded_msg = BytesMut::new();
-        item.encode(&mut encoded_msg)
-            .expect("BytesMut to have sufficient capacity.");
+        let mut writer = Writer::new(encoded_msg.as_mut());
+        item.write_message(&mut writer)
+            .expect("Encoding to succeed");
         self.uvi.encode(encoded_msg.freeze(), dst)?;
 
         Ok(())
     }
 }
 
-impl<In, Out: Message + Default> Decoder for Codec<In, Out> {
+impl<In, Out: for<'a> MessageRead<'a> + Default> Decoder for Codec<In, Out> {
```

Comment:
Have you tried to fix this to `'static`?

Reply:
That doesn't work because `msg` in `Self::Item::from_reader(&mut reader, &msg)` does not live long enough.